v0.31.4 is unable to start up

### Current Behavior

"C:\Program Files\windows_exporter\windows_exporter.exe" --config.file="C:\Program Files\windows_exporter\config.yaml" --collectors.enabled cpu,cpu_info,logical_disk,memory,net,netframework,os,pagefile,performancecounter,process,service,system,tcp,terminal_services,time,msmq,iis,mssql    --collector.process.iis --collector.process.counter-version=1
panic: reflect: Elem of invalid type process.perfDataCounterValues

goroutine 16 [running]:
reflect.elem(0xc000120101?)
        C:/hostedtoolcache/windows/go/1.25.0/x64/src/reflect/type.go:742 +0x9a
reflect.(*rtype).Elem(0x7ff65f8d2f8d?)
        C:/hostedtoolcache/windows/go/1.25.0/x64/src/reflect/type.go:746 +0x15
github.com/prometheus-community/windows_exporter/internal/pdh/registry.NewCollector[...]({0x7ff65f8d2f8d, 0x7}, {0x1, 0x0?, 0x0?})
        D:/a/windows_exporter/windows_exporter/internal/pdh/registry/collector.go:56 +0x11f
github.com/prometheus-community/windows_exporter/internal/collector/process.(*Collector).Build(0xc0000f1900, 0xc0001e0bc0, 0xc0001e2a40)
        D:/a/windows_exporter/windows_exporter/internal/collector/process/process.go:192 +0x3e5
github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build.func1()
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:230 +0x89
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Collection).Build in goroutine 1
        D:/a/windows_exporter/windows_exporter/pkg/collector/collection.go:227 +0x1f0

### Expected Behavior

The process can start up similarly to v0.31.3.

### Steps To Reproduce

```Markdown
Updated deployment from windows_exporter-0.31.3-amd64.msi to windows_exporter-0.31.4-amd64.msi, and the Windows Exporter service is unable to start.
```

### Environment

- Windows Server Version: 2022 and 2025


### windows_exporter logs

```shell
Above
```

### Anything else?

_No response_

--- a/internal/pdh/registry/collector.go
+++ b/internal/pdh/registry/collector.go
@@ -53,7 +53,7 @@ func NewCollector[T any](object string, _ []string) (*Collector, error) {
 		counters:       make(map[string]Counter),
 	}
 
-	valueType := reflect.TypeFor[T]().Elem()
+	valueType := reflect.TypeFor[T]()
 
 	if f, ok := valueType.FieldByName("Name"); ok {
 		if f.Type.Kind() == reflect.String {

--- a/internal/pdh/registry/perflib_test.go
+++ b/internal/pdh/registry/perflib_test.go
@@ -21,6 +21,22 @@ import (
 	"testing"
 )
 
+// TestNewCollectorStructTypeParam guards against a regression where
+// reflect.TypeFor[T]().Elem() panicked when T is a plain struct (not a pointer).
+// See https://github.com/prometheus-community/windows_exporter/issues/2365
+func TestNewCollectorStructTypeParam(t *testing.T) {
+	type systemCounterValues struct {
+		Name string
+
+		ProcessorQueueLength float64 `perfdata:"Processor Queue Length"`
+	}
+
+	_, err := NewCollector[systemCounterValues]("System", nil)
+	if err != nil {
+		t.Skipf("skipping: failed to create collector: %v", err)
+	}
+}
+
 func BenchmarkQueryPerformanceData(b *testing.B) {
 	for b.Loop() {
 		_, _ = QueryPerformanceData("Global", "")

